### ISSUE TITLE
Ability to skip the captured log of successful tests

### ISSUE THREAD

**Issue by sylvinus** [pytest-dev/pytest-html#105]
Hello,

We have a few thousand tests and the generated HTML file weights close to 100Mo, basically unreadable in a browser.

I've made this patch to make the file much smaller, because typically only the captured output of failures is interesting:
https://github.com/pricingassistant/pytest-html/commit/0df1e0e249cfe322592e87a3868342a401d64b1e

Could this be merged behind a new configuration flag?

**Comment by davehunt** [comment]
You want the passing tests to be omitted from the report? That sounds reasonable to me for such large result sets. Please submit your change as a patch with the configuration flag.

**Comment by davehunt** [comment]
Actually, with #104 merged and released you should be able to do this via a `conftest.py`:

```python
@pytest.mark.optionalhook
def pytest_html_results_table_row(report, cells):
    xfail = hasattr(report, 'wasxfail')
    if (report.skipped and not xfail) or (report.failed and xfail):
        cells = []
```

It's worth testing to see if this works - it could be that the we'd need to support making cells `None` and skipping the append row.

**Comment by sylvinus** [comment]
@davehunt yes that PR looks great, it would indeed be good to add control over the `append_log_html` call as well. We do want to display the successful tests, just not their whole log output.

thanks!

**Comment by davehunt** [comment]
Ah, I see. Let me see what I can come up with.

**Comment by davehunt** [comment]
@sylvinus let me know if the new commits to #104 work for you. If so, I'll hopefully be doing a release later today.

**Comment by sylvinus** [comment]
yes I think that would :) thanks!

**Comment by davehunt** [comment]
Fixed by adding hooks in bc2e8a35554df68113cd2ae911dc396cd32cfe93 available in v1.14.0.
